hal/ubuntu/qemu.sh

```
@@ -94,27 +205,76 @@ EOF
+
+if [ "$TDX_AVAILABLE" = true ]; then
+  echo "Starting QEMU VM with Intel TDX (Confidential VM)..."
+  # Update the -name option to include process and debug-threads
+  QEMU_OPTS=$(echo "$QEMU_OPTS" | sed "s/-name $VM_NAME/-name $VM_NAME,process=$VM_NAME,debug-threads=on/")
+  # Remove -m and add memory-backend-memfd for TDX (critical!)
+  QEMU_OPTS=$(echo "$QEMU_OPTS" | sed "s/-m $RAM//")
+  QEMU_OPTS="$QEMU_OPTS -object memory-backend-memfd,id=ram1,size=$RAM,share=true,prealloc=false"
+  QEMU_OPTS="$QEMU_OPTS -m $RAM"
+  QEMU_OPTS="$QEMU_OPTS -cpu host,pmu=off"
+  # TDX guest object with quote generation socket
+  QEMU_OPTS="$QEMU_OPTS -object {\"qom-type\":\"tdx-guest\",\"id\":\"tdx0\",\"quote-generation-socket\":{\"type\":\"vsock\",\"cid\":\"2\",\"port\":\"4050\"}}"
```

Shellcheck (0.11.0)

[warning] Quotes/backslashes will be treated literally. Use an array.

(SC2089)